### PR TITLE
Small parser refactor / cleanup

### PR DESCRIPTION
**Pull request by nekevss**
This PR streamlines some of the ISO8601 parsing and removes unnecessary iterations.

### CODE REVIEW

File: src/parser/time.rs [hunk 1]
```diff
@@ -66,35 +66,23 @@ pub(crate) fn parse_time_spec(cursor: &mut Cursor) -> TemporalResult<TimeSpec> {
...
+    let hour_value = first * 10 + cursor.next_digit()?;
     if !(0..=23).contains(&hour_value) {
         return Err(TemporalError::syntax().with_message("Hour must be in a range of 0-23"));
     }
     Ok(hour_value)
 }
 
 // NOTE: `TimeSecond` is a 60 inclusive `MinuteSecond`.
-/// Parse `MinuteSecond`
+/// Parse `MinuteSecond` value.
```
Comment (jedel1043) [inline src/parser/time.rs:81]: Can you link to the corresponding grammar rule?

**Review by jedel1043 — APPROVED** [review]
Nice refactor! Makes it very clean for numeric parses.

File: src/parser/time.rs (hunk 1 above)
Comment (nekevss) [reply src/parser/time.rs:81]: Can do. Better documentation all the way around is definitely needed. Should we update the grammar rules as a whole in a follow-up PR or do you mind if I add them in on this PR?

Comment (jedel1043) [reply src/parser/time.rs:81]: If you're planning on doing a PR only for documentation, you can move this change to that PR.

**Pull request merged**